sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -500,11 +501,17 @@ class ResolveSessionCatalog(
         v1TableName.asTableIdentifier,
         "ALTER TABLE RECOVER PARTITIONS")
 
-    case AlterTableAddPartitionStatement(tbl, partitionSpecsAndLocs, ifNotExists) =>
-      val v1TableName = parseV1Table(tbl, "ALTER TABLE ADD PARTITION")
+    case AlterTableAddPartition(r @ ResolvedTable(_, _, _: V1Table), partSpecsAndLocs, ifNotExists)
+        if isSessionCatalog(r.catalog) =>
       AlterTableAddPartitionCommand(
-        v1TableName.asTableIdentifier,
-        partitionSpecsAndLocs,
+        r.identifier.asTableIdentifier,
+        partSpecsAndLocs.asUnresolvedPartitionSpecs.map(spec => (spec.spec, spec.location)),
+        ifNotExists)
+
+    case AlterTableAddPartition(r: ResolvedView, partSpecsAndLocs, ifNotExists) =>
```

Comment:
We don't need to match this case if you follow my suggestion in https://github.com/apache/spark/pull/29339/files#r510688469